compiler/rustc_infer/src/infer/error_reporting/need_type_info.rs

```diff
@@ -856,3 +868,119 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
         err
     }
 }
+
+/// Turn resolved type params into `[type error]` to signal we don't want to display them.
```

Comment:
Describing here the reasoning for why we go through `[type error]` first before we immediately replace them with `TyInfer` again would be useful.

Reply:
Expanded.